src/main/java/net/sf/jabref/gui/search/GlobalSearchBar.java

```diff
@@ -16,201 +16,203 @@
 import java.util.Objects;
 import java.util.stream.Collectors;
 
+import javax.swing.AbstractAction;
 import javax.swing.JButton;
 import javax.swing.JLabel;
 import javax.swing.JPanel;
 import javax.swing.JToggleButton;
 import javax.swing.JToolBar;
 
 import net.sf.jabref.Globals;
-import net.sf.jabref.JabRefPreferences;
 import net.sf.jabref.gui.BasePanel;
 import net.sf.jabref.gui.IconTheme;
+import net.sf.jabref.gui.JabRefFrame;
 import net.sf.jabref.gui.OSXCompatibleToolbar;
 import net.sf.jabref.gui.WrapLayout;
 import net.sf.jabref.gui.autocompleter.AutoCompleteSupport;
 import net.sf.jabref.gui.help.HelpAction;
 import net.sf.jabref.gui.help.HelpFiles;
+import net.sf.jabref.gui.keyboard.KeyBinding;
 import net.sf.jabref.gui.maintable.MainTableDataModel;
 import net.sf.jabref.gui.util.component.JTextFieldWithUnfocusedText;
-import net.sf.jabref.gui.worker.AbstractWorker;
 import net.sf.jabref.logic.autocompleter.AutoCompleter;
 import net.sf.jabref.logic.l10n.Localization;
 import net.sf.jabref.logic.search.SearchQuery;
 import net.sf.jabref.logic.search.SearchQueryHighlightObservable;
 import net.sf.jabref.logic.util.OS;
 import net.sf.jabref.model.entry.BibEntry;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-
-/**
- * The search bar at the top of the screen allowing the user to search his database.
- */
-public class SearchBar extends JPanel {
-
-    private static final Log LOGGER = LogFactory.getLog(SearchBar.class);
+public class GlobalSearchBar extends JPanel {
 
+    private static final Color NEUTRAL_COLOR = Color.WHITE;
     private static final Color NO_RESULTS_COLOR = new Color(232, 202, 202);
     private static final Color RESULTS_FOUND_COLOR = new Color(217, 232, 202);
     private static final Color ADVANCED_SEARCH_COLOR = new Color(102, 255, 255);
 
-    private final JButton openCurrentResultsInDialog;
-    private final JButton globalSearch;
-    private final JButton searchModeButton;
-
-    private final BasePanel basePanel;
-
-    private final SearchQueryHighlightObservable searchQueryHighlightObservable;
-    private final JTextFieldWithUnfocusedText searchField = new JTextFieldWithUnfocusedText(
-            Localization.lang("Search") + "...");
-
-    private SearchMode searchMode = getSearchModeFromSettings();
+    private final JabRefFrame frame;
 
+    private final JToolBar toolBar = new OSXCompatibleToolbar();
+    private final JLabel searchIcon = new JLabel(IconTheme.JabRefIcon.SEARCH.getSmallIcon());
+    private final JTextFieldWithUnfocusedText searchField = new JTextFieldWithUnfocusedText(Localization.lang("Search") + "...");
+    private final JButton clearSearchButton = new JButton(IconTheme.JabRefIcon.CLOSE.getSmallIcon());
+    private final JButton openCurrentResultsInDialog = new JButton(IconTheme.JabRefIcon.OPEN_IN_NEW_WINDOW.getSmallIcon());
+    private final JButton globalSearch = new JButton(Localization.lang("Search globally"));
```

Comment:
Can you find an icon for this as well? Toolbars shouldn't include buttons with large strings.